File: HW/【前缀和】2023B-数字游戏.py
#
# @hw id=2023B lang=python3
#
# 数字游戏
#

# @hw code=start
# (sum[j] - nums[i])  = 0
# sum[j] % target = sum[i] % target

n, m = map(int,input().split())
nums = list(map(int,input().split()))
sum = 0
sumSet = set()
sumSet.add(0)
flag = 0
for num in nums:
    sum += num
    if (sum % m) in sumSet:
        flag = 1
        break
    else:
        sumSet.add(sum % m)
print(flag)
# @hw code=end

File: HW/【前缀和】2023B-数字游戏.cpp
/*
 * @hw id=2023B lang=cpp
 *
 * 数字游戏
 */

// @hw code=start
#include<iostream>
#include<set>
#include<vector>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    int nums[n];
    for(int i=0;i<n;i++) {
        cin >> nums[i];
    }
    int sum = 0;
    set<int> sumSet;
    sumSet.insert(0);
    int flag = 0;
    for(int num : nums) {
        sum += num;
        if(sumSet.count(sum%m)) {
            flag = 1;
            break;
        }
        sumSet.insert(sum%m);
    }
    cout << flag << endl;
}
// @hw code=end